### ISSUE TITLE
No Environemental data found : UNKNOWN

### ISSUE THREAD

**Issue by ste-giraldo** [DarkNinja77/check_snmp_environment.pl#2]
Trying to use this script against some ASR1001-X in IOS 17.3.4a and 16.12.3 and also against some ISR4451 in 16.9, but for all of them I always get the error: `No Environemental data found : UNKNOWN. `

SNMP ACLs in routers shows matches (2 for each trying), this means I can reach the device and also the community is right (copy paste from the device). 

What's wrong!? Except the typo of "Environemental" instead of "Environmental" :D

**Comment by DarkNinja77** [comment]
The typo you found also existed in the original code that I branched from. I fixed the typo you found and another one in my code. Thanks!

Did you use the "-T ciscoNEW" option? Try "check_snmp_environment.pl --help" to see what device types the script supports. Newer Cisco routers use a different SNMP MIB for environmental data, which is why you have to tell the script if you want to query an old Cisco "-T cisco" or a new Cisco "-T ciscoNEW".

Cisco 2800, 3800, 2900, 3900 series: use "cisco" type
Cisco ISR4K: use "ciscoNEW" type

The script works for me against a Cisco 4431 running 16.9.

The Nagios command definitions I use are:

define command {
        command_name    check_environment_cisco_isr4k
        command_line    $USER1$/check_snmp_environment.pl -2 -t 10 -H $HOSTADDRESS$ -C $USER2$ -T ciscoNEW -f -R '^T
emp: Inlet( 1)* R0/0$'
}
define command {
        command_name    check_environment_cisco_isr4321
        command_line    $USER1$/check_snmp_environment.pl -2 -t 10 -H $HOSTADDRESS$ -C $USER2$ -T ciscoNEW -f -R 'Te
mp: Internal'
}
define command {
        command_name    check_environment_cisco_isrG2
        command_line    $USER1$/check_snmp_environment.pl -2 -t 10 -H $HOSTADDRESS$ -C $USER2$ -T cisco -f -R "^Inta
ke$|^Intake (Left|Right)$|^chassis$"
}
define command {
        command_name    check_environment_cisco_switch
        command_line    $USER1$/check_snmp_environment.pl -2 -t 10 -H $HOSTADDRESS$ -C $USER2$ -T cisco -f -R "Inlet
 Temp Sensor|Sensor#1|Temp [Ss]ensor 0|^Air [Ii]nlet$|Temp: *[Ii]nlet$"
}

**Comment by ste-giraldo** [comment]
RTFM should be your answer! :D Thanks a lot, also thanks for the command definitions.

**Comment by DarkNinja77** [comment]
No worries - I'm new to GitHub and publishing code on it. A year or two ago after I coded the changes to the original check_snmp_environment.pl, I thought somebody else might also be interested in proper ISR4K support and just pasted my version of the script to GitHub.
Looks like somebody else now is using my version of the script. :-)

I just added the command definitions to the README.md, which GitHub shows on the project page. Maybe others are also interested in how I am calling the script from Nagios. I am also using PNP4Nagios to get nice graphs from collected performance data, and I only plot the temperature, which is why i filter the performance output (-f) with a regex filter (-R) to only show the (inlet) temperature of my routers.
